src/peergos/server/cli/Command.java

```diff
@@ -2,33 +2,45 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.List;
-import java.util.Set;
 import java.util.stream.Stream;
 
 public enum Command {
     help("Show this help"),
     exit("Disconnect"),
-    get("Download a file", "get remote-path <local path>"),
-    put("Upload a file", "put local-path <remote-path>"),
-    ls("List contents of a remote directory", "ls <path>"),
-    rm("Remove a remote-file", "rm remote-path"),
+    get("Download a file", "get remote-path <local path>", Arg.REMOTE, Arg.LOCAL),
+    put("Upload a file", "put local-path <remote-path>", Arg.LOCAL, Arg.REMOTE),
+    ls("List contents of a remote directory", "ls <path>", Arg.REMOTE),
+    rm("Remove a remote-file", "rm remote-path", Arg.REMOTE),
     space("Show used remote space"),
     get_follow_requests("Show the users that have sent you a follow request"),
-    follow("Send a follow-request to another user.", "follow username-to-follow"),
-    share_read("Grant read access for a file to another user.", "share_read path <user>"),
+    follow("Send a follow-request to another user.", "follow username-to-follow", Arg.USERNAME),
+    share_read("Grant read access for a file to another user.", "share_read path <user>", Arg.REMOTE, Arg.FOLLOWER),
     passwd("Update your password"),
-    cd("change (remote) directory", "cd <path>"),
+    cd("change (remote) directory", "cd <path>", Arg.REMOTE),
```

Comment:
would it make sense to split arg.Remote into dirOnly for this command? Otherwise an autcomplete in a dir with 1 subdir and 1000s of files will be awkward?

Reply:
Sure, I  think it's  a nice-to-have though  -  thousands of files in one dir is always going to be awkward.

Reply:
I think it's an important UX improvement (bash does it). It matters even is there is only 1 dir and 1 file. It's also easy to do.

Reply:
I didn't really read this the first time - yes that makes sense - now done :)